app/src/main/java/org/fossasia/openevent/general/favorite/FavoriteFragment.kt

```diff
@@ -49,6 +51,16 @@ class FavoriteFragment : Fragment() {
             thisActivity.supportActionBar?.setDisplayHomeAsUpEnabled(false)
         }
 
+        rootView.findSomethingToDo.setOnClickListener {
+            val builder = NavOptions.Builder()
+            builder.setPopUpTo(R.id.eventsFragment, false)
+            builder.setEnterAnim(R.anim.slide_in_right)
+            builder.setExitAnim(R.anim.slide_out_left)
+            builder.setPopEnterAnim(R.anim.slide_in_left)
+            builder.setPopExitAnim(R.anim.slide_out_right)
```

Comment:
The builder pattern was created so that people don't have to write `builder.` for each function call

Reply:
@iamareebjamal Should I change something, because I cannot find any predefined method for the coded part.

Reply:
Write - 
```java
val navOptions = NavOptions.Builder()
    .setPopUpTo(R.id.eventsFragment, false)
    .setEnterAnim(R.anim.slide_in_right)
    .setExitAnim(R.anim.slide_out_left)
    .setPopEnterAnim(R.anim.slide_in_left)
    .setPopExitAnim(R.anim.slide_out_right)
    .build()
Navigation.findNavController(rootView).navigate(R.id.searchFragment, null, navOptions)
```

Reply:
@liveHarshit @iamareebjamal How is this code snippet different from the existing one ? Am I missing something here ?

Reply:
Yes, use a diffing tool to see what's different